clients/common/norm.cpp

```diff
@@ -44,122 +46,135 @@
 extern "C" {
 #endif
 
-float  slange_(char* norm_type, int* m, int* n, float* A, int* lda, float* work);
-double dlange_(char* norm_type, int* m, int* n, double* A, int* lda, double* work);
-float  clange_(char* norm_type, int* m, int* n, hipblasComplex* A, int* lda, float* work);
-double zlange_(char* norm_type, int* m, int* n, hipblasDoubleComplex* A, int* lda, double* work);
+// float  slange_(char* norm_type, int* m, int* n, float* A, int* lda, float* work);
+// double dlange_(char* norm_type, int* m, int* n, double* A, int* lda, double* work);
+// float  clange_(char* norm_type, int* m, int* n, hipblasComplex* A, int* lda, float* work);
+// double zlange_(char* norm_type, int* m, int* n, hipblasDoubleComplex* A, int* lda, double* work);
 
-float  slansy_(char* norm_type, char* uplo, int* n, float* A, int* lda, float* work);
-double dlansy_(char* norm_type, char* uplo, int* n, double* A, int* lda, double* work);
+// float  slansy_(char* norm_type, char* uplo, int* n, float* A, int* lda, float* work);
+// double dlansy_(char* norm_type, char* uplo, int* n, double* A, int* lda, double* work);
 //  float  clanhe_(char* norm_type, char* uplo, int* n, hipblasComplex* A, int* lda, float* work);
 //  double zlanhe_(char* norm_type, char* uplo, int* n, hipblasDoubleComplex* A, int* lda, double*
 //  work);
 
-void saxpy_(int* n, float* alpha, float* x, int* incx, float* y, int* incy);
-void daxpy_(int* n, double* alpha, double* x, int* incx, double* y, int* incy);
-void caxpy_(
-    int* n, hipblasComplex* alpha, hipblasComplex* x, int* incx, hipblasComplex* y, int* incy);
-void zaxpy_(int*                  n,
-            hipblasDoubleComplex* alpha,
-            hipblasDoubleComplex* x,
-            int*                  incx,
-            hipblasDoubleComplex* y,
-            int*                  incy);
+// void m_axpy_64(int* n, float* alpha, float* x, int* incx, float* y, int* incy);
+// void m_axpy_64(int* n, double* alpha, double* x, int* incx, double* y, int* incy);
+// void m_axpy_64(
+//     int* n, hipblasComplex* alpha, hipblasComplex* x, int* incx, hipblasComplex* y, int* incy);
+// void m_axpy_64(int*                  n,
+//             hipblasDoubleComplex* alpha,
+//             hipblasDoubleComplex* x,
+//             int*                  incx,
+//             hipblasDoubleComplex* y,
+//             int*                  incy);
 
 #ifdef __cplusplus
 }
 #endif
 
+template <typename T>
+void m_axpy_64(int64_t N, T* alpha, T* x, int64_t incx, T* y, int64_t incy)
```

Comment:
Why do we need this one here?

Reply:
What do you mean exactly, why not reuse ref_  form?   It alllows a self contained 64bit support in norm.cpp.  Windows will get additions to allow 32bit but I think linux may be stuck with always 64bit reference.   Due to variations it maybe only solver -> LAPACKE will be wrapped with #ifdef WIN32 to allow 32bit API visibility.